### PR TITLE
[WIP] Medium Roleplay Traitors (Lots of fluff, refreshing objectives on completion.)

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request
besides getting a lot of fluff when starting and being distinctly "terrorist factions" when they need to hijack, traitors now get the ability to request more objectives when they are done with all of their objectives. they can repeat this every 10 minutes to get more, as long as they still have been doing their objectives.

what's left to do in the pr?
- [ ] make the gamemode
- [ ] make the dynamic entry
- [ ] remove testing cooldowns (currently 5 seconds to try again for objectives, that needs to be high

## Why It's Good For The Game

taken from the code (because it's correct and succinct)
//1. Added a lot of fluff, this lets people build and play a better character
//2. Gives them refreshing objectives so they can continually antagonize the round.

to expand on the second point: traitors on manuel have trouble actually properly being an active threat to the station. There's a chance they're not going to antagonize but instead leave, and there's a chance they do all their objectives and then become a "completed" threat for the station. in not getting caught they closed all need to actually antagonize further and so they just kind of stop. They also have to stop, since any more would be murderbone. Why not keep them coming back with new targets if they are sufficiently killing people? Contracting does this well and it worked there, so i think it's the push manuel needs to have a bit more action.

## Changelog
:cl:
add: Medium Roleplay Traitors.
/:cl:

### CODE REVIEW

**Comment by tralezab** [comment]
These will be disabled on all servers by default, headmins need to enable them.

**Comment by Tlaltecuhtli** [comment]
can you add isis?

**Comment by wabash1** [comment]
Nerds on bagil like reasons to do more shit too. Though we may want some sort of incentive like getting more TC for each new completed objective.

**Comment by itseasytosee** [comment]
I would check out [https://github.com/tgstation/common_core/blob/master/Syndicate.md](url) for lore inspiration.

**Comment by ATH1909** [comment]
> Nerds on bagil like reasons to do more shit too. Though we may want some sort of incentive like getting more TC for each new completed objective.

I feel like giving TC for each completed objective would both step on the turf of contractors and be really hard to implement for things like theft objectives (or even "don't let X escape alive" objectives).

Making it TC-less, like this PR, is fine. Question, though- are things like stealing an SM sliver in the pool of possible objectives?

File: code/modules/antagonists/traitor/medrp/medrp_traitor.dm [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+			. += "<span class='notice'>The creatures of this world must be freed from the iron grasp of Nanotrasen, and you are their only hope!</span>"
+			. += "<span class='notice'>The Syndicate have graciously given one of their uplinks for your task.</span>"
+		if("Bee Liberation Front")
+			. += "<span class='userdanger'>You are the Bee Liberation Front Operative.</span>"
+			. += "<span class='big bold'>You may cooperate with other syndicate operatives if they support our cause. Maybe you can recruit an Animal Rights Consort to be useful for once?</span>"
+			. += "<span class='notice'>We must prove ourselves to the Syndicate or we will not be able to join. Animal Rights Consort will roll us!</span>"
+			. += "<span class='notice'>The Syndicate have graciously given one of their uplinks to see if we are worthy.</span>"
+		if("Cybersun Industries")
+			. += "<span class='userdanger'>You are from Cybersun Industries.</span>"
+			. += "<span class='big bold'>All other syndicate operatives are not to be trusted. Fellow Cybersun operatives are to be trusted. Members of the MI13 organization can be trusted.</span>"
```
Comment (itseasytosee) [inline code/modules/antagonists/traitor/medrp/medrp_traitor.dm:75]:
```suggestion
			. += "<span class='big bold'>Fellow Cybersun operatives are to be trusted. Members of the MI13 organization can be trusted. All other syndicate operatives are not to be trusted. </span>"
```
I know this is how its written on the wiki but its bad grammar

**Comment by JohnFulpWillard** [comment]
I love this, maybe in the future we can add Faction-limited traitor items, like how Tiger Co is supposed to be using more old-school but deadly weapons, while Cybersun could maybe get the Illegal HUD that headrevs get, in an autosurgeon (MI13 too)

**Comment by Tlaltecuhtli** [comment]
> > Nerds on bagil like reasons to do more shit too. Though we may want some sort of incentive like getting more TC for each new completed objective.
> 
> I feel like giving TC for each completed objective would both step on the turf of contractors and be really hard to implement for things like theft objectives (or even "don't let X escape alive" objectives).
> 
> Making it TC-less, like this PR, is fine (and I could even see this being expanded to be available to all antag types). Question, though- are things like stealing an SM sliver in the pool of possible objectives?

what if we remove fetch quest obj? contractors is just better traitors

**Comment by wabash1** [comment]
Why not an objective pinpointer for fetch quest objects for 1-2 TC?  Fetch
quest should probably not be in the list of new quests given it'll usually
be go get the hos gun that the other syndicate traitor already got.

**Comment by DatBoiTim** [comment]
Do they get additional TC upon further objective requests? I can see people getting stretched thin later on.

**Comment by ArcaneMusic** [comment]
I personally lack the know-how to recommend HOW to do this, but you might consider throwing the antagonist strings into a .json file for ease of use, and parsing it out of that.

**Comment by tralezab** [comment]
I knew json was gonna come up, yeah I prob could in some way

File: code/modules/antagonists/traitor/medrp/new_objectives.dm [hunk 2]
```diff
@@ -0,0 +1,34 @@
+
+
+/obj/effect/proc_holder/spell/self/add_objectives //If all of your objectives are done, you get two more.
+	name = "Request More Objectives"
```
Comment (tralezab) [inline code/modules/antagonists/traitor/medrp/new_objectives.dm:4]: At the moment I liked how sticky spells were in staying attached to the character (aka it follows mind, not mob) but actually a way better idea is making this a button in an uplink.

File: code/modules/antagonists/traitor/medrp/medrp_traitor.dm (hunk 1 above)
Comment (tralezab) [reply code/modules/antagonists/traitor/medrp/medrp_traitor.dm:75]: Sure

File: code/modules/antagonists/traitor/medrp/medrp_traitor.dm [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+	switch(role_title)
+		if("Tiger Cooperative")
+			. += "<span class='userdanger'>You are the Tiger Cooperative Fanatic.</span>"
+			. += "<span class='big bold'>Only the enlightened Tiger brethren can be trusted; all others must be expelled from this mortal realm!</span>"
+			. += "<span class='notice'>Remember the teachings of Hy-lurgixon; kill first, ask questions later!</span>"
+			. += "<span class='notice'>You have been provided with a standard uplink to prove yourself to the changeling hive. If you accomplish your tasks, you will be assimilated.</span>"
+		if("Waffle Corporation Terrorist")
+			. += "<span class='userdanger'>You are the Waffle Corporation Terrorist.</span>"
+			. += "<span class='big bold'>Most other syndicate operatives are not to be trusted, except for members of the Gorlex Marauders. Do not trust fellow members of the Waffle.co (but try not to rat them out), as they might have been assigned opposing objectives.</span>"
+			. += "<span class='notice'>It has been a relatively quiet month, many of our black market weapons have gone unused. Let's give Nanotrasen a live test and hijack one of their shuttles.</span>"
```
Comment (tralezab) [inline code/modules/antagonists/traitor/medrp/medrp_traitor.dm:62]: unhappy with this line specifically

**Comment by optimumtact** [comment]
this is just traitor, so I don't really understand the point

**Comment by tralezab** [comment]
I'll apply them to all traitors then, sure

File: code/modules/antagonists/traitor/medrp/medrp_traitor.dm [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+			. += "<span class='notice'>You have been provided with a standard uplink to accomplish your task.</span>"
+		if("Animal Rights Consortium")
+			. += "<span class='userdanger'>You are the ARC Terrorist.</span>"
+			. += "<span class='notice'>You may cooperate with other syndicate operatives if they support our cause. Maybe you can convince the Bee Liberation Front operatives to cooperate for once?</span>"
+			. += "<span class='bold'>The creatures of this world must be freed from the iron grasp of Nanotrasen, and you are their only hope!</span>"
+			. += "<span class='notice'>The Syndicate have graciously given one of their uplinks for your task.</span>"
+		if("Bee Liberation Front")
+			. += "<span class='userdanger'>You are the Bee Liberation Front Operative.</span>"
+			. += "<span class='notice'>You may cooperate with other syndicate operatives if they support our cause. Maybe you can recruit an Animal Rights Consort to be useful for once?</span>"
+			. += "<span class='bold'>We must prove ourselves to the Syndicate or we will not be able to join. Animal Rights Consort will roll us!</span>"
```
Comment (ATH1909) [inline code/modules/antagonists/traitor/medrp/medrp_traitor.dm:82]: that second sentence might need a revision

File: code/modules/antagonists/traitor/medrp/medrp_traitor.dm [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+		if("Waffle Corporation Terrorist")
+			. += "<span class='userdanger'>You are the Waffle Corporation Terrorist.</span>"
+			. += "<span class='notice'>Most other syndicate operatives are not to be trusted, except for members of the Gorlex Marauders. Do not trust fellow members of the Waffle.co (but try not to rat them out), as they might have been assigned opposing objectives.</span>"
+			. += "<span class='bold'>An egregiously loud feat like hijacking a Nanotrasen emergency shuttle should attract customers.</span>"
+			. += "<span class='notice'>You have been provided with a standard uplink to accomplish your task.</span>"
+		if("Animal Rights Consortium")
+			. += "<span class='userdanger'>You are the ARC Terrorist.</span>"
+			. += "<span class='notice'>You may cooperate with other syndicate operatives if they support our cause. Maybe you can convince the Bee Liberation Front operatives to cooperate for once?</span>"
+			. += "<span class='bold'>The creatures of this world must be freed from the iron grasp of Nanotrasen, and you are their only hope!</span>"
+			. += "<span class='notice'>The Syndicate have graciously given one of their uplinks for your task.</span>"
```
Comment (ATH1909) [inline code/modules/antagonists/traitor/medrp/medrp_traitor.dm:78]: is this meant to be "given **us**", not just "given"?

File: code/modules/antagonists/traitor/medrp/new_objectives.dm [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+	desc = "Grants you more objectives if every objective you have right now is complete."
+	still_recharging_msg = "<span class='notice'>You need to wait for awhile between objective requests.</span>"
+	clothes_req = FALSE
+	antimagic_allowed = TRUE
+	charge_max	= 10 MINUTES
+	charge_counter = 0
+
+/obj/effect/proc_holder/spell/self/add_objectives/cast(list/targets, mob/living/carbon/human/user)
+
+	var/datum/antagonist/traitor/roleplay/antag_datum = user.mind.has_antag_datum(/datum/antagonist/traitor/roleplay)
```
Comment (ATH1909) [inline code/modules/antagonists/traitor/medrp/new_objectives.dm:14]: it'd be nice if we could turn this into a generic thing that could work with any antag type (with the right subtypes) at some point.

File: code/modules/antagonists/traitor/medrp/medrp_traitor.dm [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+			else
+				return pickweight(list("Cybersun Industries" = 1, "MI13" = 1, "Gorlex Marauders" = 1, "Donk Corporation" = 1, "Waffle Corporation" = 1))
+		if("Nanotrasen")
+			if(hijacking)
+				return "Gone Postal"
+			else
+				return pickweight(list("Internal Affairs Agent" = 3, "Corporate Climber" = 1, "Legal Trouble" = 1))
+
+//moved to the bottom of the file because of it's size
+/datum/antagonist/traitor/roleplay/proc/fluff_strings()
```
Comment (ATH1909) [inline code/modules/antagonists/traitor/medrp/medrp_traitor.dm:61]: do these messages get put in your notes tab as well?

File: code/modules/antagonists/traitor/medrp/medrp_traitor.dm [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+	. = list()
+	switch(role_title)
+		if("Tiger Cooperative")
+			. += "<span class='userdanger'>You are the Tiger Cooperative Fanatic.</span>"
```
Comment (optimumtact) [inline code/modules/antagonists/traitor/medrp/medrp_traitor.dm:65]: change these to loading from config string files

this is lazy coding

File: code/modules/antagonists/traitor/medrp/medrp_traitor.dm [hunk 9]
```diff
@@ -0,0 +1,129 @@
...
+		var/obj/effect/proc_holder/spell/self/add_objectives/removed_spell = s
+		if(istype(removed_spell))
+			owner.RemoveSpell(removed_spell)
+
+/datum/antagonist/traitor/roleplay/proc/handle_employer()
+	hijacking = locate(/datum/objective/hijack) in objectives
+	switch(employer)
+		if("Syndicate")
+			if(hijacking)
+				return pickweight(list("Tiger Cooperative Fanatic" = 3, "Waffle Corporation Terrorist" = 1, "Animal Rights Consortium" = 1,"Bee Liberation Front" = 1))
```
Comment (optimumtact) [inline code/modules/antagonists/traitor/medrp/medrp_traitor.dm:51]: the list of traitor factions should also be configurable by server operators

File: code/modules/antagonists/traitor/medrp/new_objectives.dm [hunk 10]
```diff
@@ -0,0 +1,36 @@
+
+
+/obj/effect/proc_holder/spell/self/add_objectives //If all of your objectives are done, you get two more.
```
Comment (optimumtact) [inline code/modules/antagonists/traitor/medrp/new_objectives.dm:3]: THIS SHOULD NOT BE A SPELL, ABSOLUTELY DISGRACEFUL, LEARN HOW TO ADD ACTION BUTTONS

**Review by optimumtact — CHANGES_REQUESTED** [review]
Terribly lazy coding here, not acceptable at all

you should know better

**Comment by Pumpkinoe** [comment]
In the arms of an angel :(

**Pull request closed without merging**